server/src/server.ts

```diff
@@ -166,9 +166,15 @@ documents.onDidOpen(analyzeDocument)
 documents.onDidChangeContent(analyzeDocument)
 
 documents.onDidSave(async (event) => {
+  logger.info(`[onDidSave] Document saved: ${event.document.uri}`)
   if (parseOnSave && !eSDKMode) {
-    logger.debug('[onDidSave] Parsing all recipes...')
-    void connection.sendRequest('bitbake/parseAllRecipes')
+    const uri = event.document.uri
+    if (path.extname(uri) === '.bb') {

```

Comment:
we could extend to more bitbake extensions: .bbappend, .inc (not .bbclass and .conf).
Then try to resolve to the name of a recipe. If it exists, scan it, if not, run the default parse all recipes task. We have the list of recipes in the project scanner. 
We want final values in .bbappend and .inc as well. I don't think they are supported yet?
You could also just pass a recipe name instead of a URI to that end.

Reply:
Updated, now it should trigger `-e` on `.bbappend` and `.inc` as well. check https://github.com/yoctoproject/vscode-bitbake/pull/92/files#diff-4398e3b2ab3e7e7e70a93409fa893abf68608d45a5ad1ac7faaf888dac77005dR172

I didn't change how the scan results are stored, so every `.bb`, `.bbappend` and `.inc` of the same recipe will have its own scan result to look up. It won't be a problem in terms of the space If the symbols in these files are mostly different because I only store the existing symbols.

There is a little duplicated code from the `selectRecipe`, but it is needed since we want unresolved recipes to trigger `parseAllRecipe`.